storage/pin/pin.go

```diff
@@ -42,23 +42,24 @@ var (
 )
 
 // FileInfo is the struct that stores the information about pinned files
-// A map of this is stored in the state DB
+// This is stored in the state DB with Address as key
 type FileInfo struct {
-	isRaw      bool
-	fileSize   uint64
-	pinCounter uint64
+	Address    storage.Address
+	IsRaw      bool
+	FileSize   uint64
+	PinCounter uint64
 }
 
 // MarshalBinary encodes the FileInfo object in to a binary form for storage
 func (f *FileInfo) MarshalBinary() (data []byte, err error) {
 	data = make([]byte, 17)
-	if f.isRaw {
```

Comment:
exposing these fields is not necessary if you implement the `MarshalJSON`/`UnmarshalJSON` methods

Reply:
DO we really have to have MarshalBinary and MarshalJSON..?

Reply:
Maybe just to add explicit json struct field tags to FileInfo would be just fine? That would fix the json encoding regardless of future field name changes (exposed api vs internal implementation). And also it would make clear that this struct is represented as json.

Reply:
@jmozah I don't know if we need both. You implemented the pin API... not me... I think we might need both but you'd be the judge of that.
In any case, adding an explicit `MarshalJSON` method gives clarity and I'm not sure if this is such a costly operation to add this to the codebase.

Reply:
@acud what would be the benefit of MarshalJSON compared to struct tags in this case?

Reply:
I any way need the fields to be exported to test in server_test.go.

Reply:
@janos I think that exporting fields in a struct affects the mutability of the data structure. I am kind of neutral on this. You can export them if you feel this is more clean.